impress-org/givewp
# Donation test data can now be generated.

Resolves #5412

## Description



Introduces a shared framework for generating fake data for GiveWP, including an abstract Factory class and custom fake data Providers.

Includes base implementations of Donors and Donations for use in Revenue Reporting.

Introduces the `give test-donors` and `give test-donations` WP CLI commands with support for flags `count` and `preview`.

## Visuals

![image](https://user-images.githubusercontent.com/10858303/97752726-f91bd980-1aca-11eb-856e-4cd2561da2d2.png)

## Pre-review Checklist




## Testing Instructions



On a clean install of WordPress and GiveWP:

1. Create a handful of donation forms.
1. Run `give test-donors --count=100` to create donors.
1. Run `give test-donations --count=100` to create donations.

- Donors should be listed in `wp_give_donors` with meta in `wp_give_donormeta`.
- Donations should be listed in `wp_posts` (`give-payment`) with meta in `wp_give_donationmeta`.
- Donations should be associated with available donation forms.
- Donations should be associated with available donors.
